torch/__init__.py

```diff
@@ -384,6 +392,20 @@ def __sym_not__(self) -> "SymBool":
     def __repr__(self):
         return str(self.node)
 
+    def __eq__(self, other):
+        if self.node.is_constant():
+            if isinstance(other, builtins.bool):
+                return self.node.bool_() == other
+            elif isinstance(other, SymBool) and other.node.is_constant():
+                return self.node.bool_() == other.node.bool_()
+        return self is other
```

Comment:
No, this is not OK. Equality needs to induce a guard, not do symbolic equality

Reply:
Going to think about this more, but I think you mean the reverse? AFAIK, we only induce guards when we do `guard_bool`, etc.

Reply:
Soo yeah, this is preexisting, i.e. `symbool == symbool` has always produced a plain bool since we never implemented `__eq__` on SymBool and the default behavior for eq is to do `is`.

Maybe we could implement `__eq__` for real in SymBool in a follow up?

Reply:
I don't expect fixing it here to have much blast radius. If you do want to do this separately, the fix diff should stack before this one.